data_steward/cdr_cleaner/cleaning_rules/convert_pre_post_coordinated_concepts.py

```diff
@@ -0,0 +1,256 @@
+SELECT DISTINCT o.* FROM `{{project}}.{{dataset}}.observation` o
+JOIN `{{project}}.{{dataset}}.concept` c1 ON o.value_source_concept_id = c1.concept_id
+JOIN `{{project}}.{{dataset}}.concept` c2 ON o.value_source_value = c2.concept_code
+JOIN `{{project}}.{{dataset}}.concept_relationship` cr1 ON c2.concept_id = cr1.concept_id_1
+WHERE c1.standard_concept IS NULL AND cr1.relationship_id = 'Maps to value'
+""")
+
+DELETE_QUERY = JINJA_ENV.from_string("""
+DELETE FROM `{{project}}.{{dataset}}.observation`
+WHERE observation_id IN (
+    SELECT observation_id FROM `{{project}}.{{sandbox_dataset}}.{{sandbox_table}}`
+)
+""")
+
+INSERT_QUERY = JINJA_ENV.from_string("""
+INSERT INTO `{{project}}.{{dataset}}.observation`
+({{observation_fields}})
+SELECT 
+    ROW_NUMBER() OVER(
+        PARTITION BY o.observation_id
+        ORDER BY
+            m.new_observation_concept_id, 
+            m.new_value_as_cocnept_id
+        ) * 100000000000 + o.observation_id AS observation_id,
+    o.person_id,
+    m.new_observation_concept_id AS observation_concept_id,
+    o.observation_date,
+    o.observation_datetime,
+    o.observation_type_concept_id,
+    o.value_as_number,
+    o.value_as_string,
+    m.new_value_as_cocnept_id AS value_as_concept_id,
+    o.qualifier_concept_id,
+    o.unit_concept_id,
+    o.provider_id,
+    o.visit_occurrence_id,
+    o.visit_detail_id,
+    o.observation_source_value,
+    o.observation_source_concept_id,
+    o.unit_source_value,
+    o.qualifier_source_value,
+    o.value_source_concept_id,
+    o.value_source_value, 
+    o.questionnaire_response_id
+FROM `{{project}}.{{sandbox_dataset}}.{{sandbox_table}}` o
+JOIN `{{project}}.{{dataset}}.concept` c
+ON o.value_source_value = c.concept_code
+JOIN `{{project}}.{{sandbox_dataset}}.{{sandbox_table}}_mapping` m
+ON c.concept_id = m.concept_id
+""")
+
+
+class ConvertPrePostCoordinatedConcepts(BaseCleaningRule):
+
+    def __init__(self,
+                 project_id,
+                 dataset_id,
+                 sandbox_dataset_id,
+                 table_namer=None):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Converts the observation records with pre-coordinated mapping to'
+            'post-coordinated mapping.')
+
+        super().__init__(issue_numbers=JIRA_ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[cdr_consts.RDR],
+                         affected_tables=[OBSERVATION],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         depends_on=[SetConceptIdsForSurveyQuestionsAnswers],
```

Comment:
I would suggest expanding the depends on to include the `UpdateFamilyHistoryCodes` rule.

Reply:
Added `UpdateFamilyHistoryCodes` as well :)